src/common/util/game.js

```diff
@@ -257,12 +257,31 @@ export function logAction(state, player, action, cards, timestamp, target = null
   return state;
 }
 
-export function newGame(state, player, usernames, decks, seed = 0) {
-  state = Object.assign(state, cloneDeep(defaultState), {player: player, rng: seededRNG(seed)}); // Reset game state.
-  state.usernames = usernames;
-  state.players.blue = bluePlayerState(decks.blue);
-  state.players.orange = orangePlayerState(decks.orange);
-  state.started = true;
+// gameMode can be 'normal' or 'sharedDeck'
+export function newGame(state, player, usernames, decks, seed = 0, gameMode = DEFAULT_GAME_MODE) {
+  // Reset game state.
+  state = Object.assign(state, cloneDeep(defaultState), {
+    gameMode,
+    player: player,
+    rng: seededRNG(seed),
+    started: true,
+    usernames
+  });
+
+  if (gameMode === 'normal') {
+    state.players.blue = bluePlayerState(decks.blue);
+    state.players.orange = orangePlayerState(decks.orange);
+  } else if (gameMode === 'sharedDeck') {
+    const deck = shuffle([...decks.blue, ...decks.orange]);
+    // Give blue the top two cards, orange the next two (to form their starting hands),
+    // and both players the rest of the deck.
+    const [topTwo, nextTwo, restOfDeck] = [deck.slice(0, 2), deck.slice(2, 4), deck.slice(4)];
+    state.players.blue = bluePlayerState([...topTwo, ...restOfDeck]);
```

Comment:
Side note for a future refactor, I think `bluePlayerState` needs to be made into a verb phrase. Like `makeBluePlayerState` right now if I look at the function name, it is unclear if I'm making an object, or checking a value on my params (i.e. is it the blue player's turn, etc.